components/builder-protocol/protocols/originsrv.proto

```diff
@@ -413,6 +413,61 @@ message OriginProjectList {
   repeated string names = 1;
 }
 
+message  OriginPublicEncryptionKey {
+  optional uint64 id = 1;
+  optional uint64 origin_id = 2;
+  optional string name = 3;
+  optional string revision = 4;
+  optional bytes body = 5;
+  optional uint64 owner_id = 6;
+}
+
+message  OriginPublicEncryptionKeyCreate {
+  optional uint64 origin_id = 1;
+  optional string name = 2;
+  optional string revision = 3;
+  optional bytes body = 4;
+  optional uint64 owner_id = 5;
+}
+
+message  OriginPublicEncryptionKeyGet {
+  optional uint64 owner_id = 1;
+  optional string origin = 2;
+  optional string revision = 3;
+}
+
+message  OriginPublicEncryptionKeyLatestGet {
+  optional uint64 owner_id = 1;
+  optional string origin = 2;
+}
+
+message OriginPublicEncryptionKeyListRequest {
+  optional uint64 owner_id = 1;
+  optional uint64 origin_id = 2;
+}
+
+message OriginPublicEncryptionKeyListResponse {
+  optional uint64 origin_id = 1;
+  repeated OriginPublicEncryptionKey keys = 2;
+}
+
+message OriginEncryptionKey {
+  optional uint64 id = 1;
+  optional uint64 origin_id = 2;
+  optional string name = 3;
+  optional string revision = 4;
+  optional bytes body = 5;
+  optional uint64 owner_id = 6;
+}
+
+message OriginEncryptionKeyCreate {
+  optional uint64 origin_id = 1;
+  optional string name = 2;
+  optional string revision = 3;
+  optional bytes body = 4;
+  optional uint64 owner_id = 5;
+}
```

Comment:
This is not a huge deal, but I'm a fan of the pattern where message that create things hold a reference to the thing they're creating, rather than duplicating all the fields.  So in this instance, `OriginEncryptionKeyCreate` would have 1 field:  `optional OriginEncryptionKey encryption_key = 1;`.  The same applies to `OriginPublicEncryptionKey` above.  Like I said, not a deal breaker, but it feels better than having all the fields duplicated.

Reply:
👍